src/aap_eda/tasks/ruleset.py

```diff
@@ -158,6 +176,18 @@ def restart(activation_id: int, requester: str = "User") -> None:
         _schedule_activate(activation, requester)
 
 
+def _job_in_queue(queue: Queue, job_id: str) -> Optional[Job]:
+    job = queue.fetch_job(job_id)
+    if job and job.get_status(refresh=True) in [
+        JobStatus.QUEUED,
+        JobStatus.STARTED,
+        JobStatus.DEFERRED,
+        JobStatus.SCHEDULED,
```

Comment:
@bzwei Since we are cancelling the RQ job before we call deactivation, if a job is being Canceled before the deactivation has started wouldn't this code let it thru and then kick off an activation while the deactivation is still running.

Reply:
This work does not address the problem you raised, nor does it make it worse. Since activation is normally slower than deactivation, activating a new instance is likely safe while an old instance is deactivating.

Reply:
Not related with this PR but I think is a fair question. Should we cancel the task before deactivation? It seems wrong to me and it is a place for race conditions, I feel the task should be cancel at the end.